### PR TITLE
[tracing] Fix null/undefined handling and be more resilient in general

### PR DESCRIPTION
**Pull request by geropl**
## Description
Fixes null/undefined handling bug and adds a generic "catch-all".

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: components/gitpod-protocol/src/util/tracing.ts
Comment (AlexTugarev) [inline components/gitpod-protocol/src/util/tracing.ts:144]: nit: not exactly, but ok. (typeof undedinfed === 'undefined')

Comment (AlexTugarev) [reply components/gitpod-protocol/src/util/tracing.ts:144]: http://wiki.ecmascript.org/doku.php?id=harmony:typeof_null

File: components/gitpod-protocol/src/util/tracing.ts
Comment (jankeromnes) [inline components/gitpod-protocol/src/util/tracing.ts:147]: I think you could also use `instanceof`, which is generally considered much safer than `typeof`:

```suggestion
                if (v instanceof Object) {
```

Screenshot 2021-12-08 at 12 08 19

File: components/gitpod-protocol/src/util/tracing.ts
Comment (AlexTugarev) [inline components/gitpod-protocol/src/util/tracing.ts:153]: this will work 👍🏻

File: components/gitpod-protocol/src/util/tracing.ts
Comment (geropl) [reply components/gitpod-protocol/src/util/tracing.ts]: Thx @jankeromnes ! My :brain: refuses to learn this JS `intricacies` so I'm doomed to re-learn every other year. :laughing:

File: components/gitpod-protocol/src/util/tracing.ts
Comment (geropl) [reply components/gitpod-protocol/src/util/tracing.ts]: Took over Jan's proposal: https://github.com/gitpod-io/gitpod/pull/7118#pullrequestreview-826303507

**Comment by geropl** [comment]
/approve no-issue

/werft run

:point_up: athens cache issue

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7118#issuecomment-988758684" title="Approved">geropl</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7118#issuecomment-988758684" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
/werft run

cleared athens, so another try...

**Comment by geropl** [comment]
/werft run


...and another one.

**Comment by geropl** [comment]
/werft run

:crossed_fingers:

**Comment by geropl** [comment]
/werft run

...

**Comment by geropl** [comment]
@jankeromnes quick, can you lgtm before the build fails again? :worried: :crossed_fingers: :stuck_out_tongue_closed_eyes:

**Review by jankeromnes — APPROVED** [review]
Looks good to me! 🧙

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 4aa079c11b85959a3020de0e7d9361e8bb285a71</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7118#issuecomment-988758684" title="Approved">geropl</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7118#pullrequestreview-826473099" title="Approved">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/7118#issuecomment-988758684" title="Approved">geropl</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
